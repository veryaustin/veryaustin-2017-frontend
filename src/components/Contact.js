import React from 'react';

const Contact = () => {
  return (
    <div id="contact">This is the Contact component</div>
  );
};

export default Contact;