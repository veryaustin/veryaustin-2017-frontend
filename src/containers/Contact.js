import React, {Component, PropTypes} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {fetchContact} from '../actions/contactActions';

class Contact extends Component {
  // Ajax call to get API data before the component mounts
  componentWillMount() {
    this.props.fetchContact("contact");
  }
  
  // Helper function to render the various ways to contact
  renderContacts(contactInfoData) {
    // Destructure the contactInfo array
    const {icon_url, site_url} = contactInfoData;
    return (
      <div key={icon_url}>
        <h5>{icon_url}</h5>
        <h5>{site_url}</h5>
      </div>
    );
  }

  render() {
    // Destructure the contact object
    const {title, slogan, contactInfo} = this.props.contact;
    if (!contactInfo) {
      return (
        <div>Loading</div>
      );
    }
    return(
      <div id="contact">
        <h1>{title}</h1>
        <h3>{slogan}</h3>
        {contactInfo.map(this.renderContacts)}
      </div>
    );
  }
}

// Map State To Props
function mapStateToProps({contact}) {
  return {contact};
}

// Bind the actions and dispatch them
function mapDispatchToProps(dispatch) {
  return bindActionCreators({fetchContact}, dispatch);
}

// Contact Props Validation
Contact.propTypes = {
  fetchContact: PropTypes.func,
  contact: PropTypes.object
};

export default connect(mapStateToProps, mapDispatchToProps)(Contact);