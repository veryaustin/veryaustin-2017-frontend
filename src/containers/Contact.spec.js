import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import configureStore from "redux-mock-store";
import Contact from "./Contact";

describe("Contact Component", () => {
  let middlewares, mockStore, store;
  beforeEach(() => {
    middlewares = [];
    mockStore = configureStore(middlewares);
  });

  it("renders the work component", () => {
    store = mockStore({
      pages: {
        contact: {}
      },
      fetching: {
        isLoading: false,
        error: null
      }
    });
    const wrapper = shallow(<Contact store={store} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the loading component when loading", () => {
    store = mockStore({
      pages: {
        contact: {}
      },
      fetching: {
        isLoading: true,
        error: null
      }
    });
    const wrapper = shallow(<Contact store={store} />);
    const loading = wrapper.find("Loading />");
    expect(loading).toExist;
  });
});
