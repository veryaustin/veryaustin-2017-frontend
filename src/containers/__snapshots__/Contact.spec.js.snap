exports[`Contact Component renders the work component 1`] = `
<Contact
  contact={Object {}}
  fetchContact={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  } />
`;
